ursh/indico_ursh/plugin.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+from __future__ import unicode_literals
+
+from flask_pluginengine import render_plugin_template
+from wtforms.fields.core import StringField
+from wtforms.fields.html5 import URLField
+from wtforms.validators import DataRequired
+
+from indico.core.plugins import IndicoPlugin, PluginCategory
+from indico.web.forms.base import IndicoForm
+from indico.web.views import WPBase
+
+from indico_ursh import _
+from indico_ursh.blueprint import blueprint
+
+
+class SettingsForm(IndicoForm):
+    api_key = StringField(_('API key'), [DataRequired()],
+                          description=_('The API key to access the <tt>ursh</tt> service'))
+    server = URLField(_('Shortening server'), [DataRequired()],
+                      description=_('The <tt>ursh</tt> shortening server (which generates short URLs)'))
+    host = URLField(_('Redirection host'), [DataRequired()],
```

Comment:
We could make this optional and use the `server` if it's not set

Reply:
Maybe, but in the case of `ursh` this does not make sense - except if we use the host of the `server`.

Reply:
It is possible to have API and redirector on the same host

Reply:
What I mean is that API will probably be under another blueprint (`/urls/` in our case). So is it safe to just drop the path and only use the host for redirection?

Reply:
Thinking about this again, why do we need the host setting at all? I think ursh should return an absolute URL when creating a new shorturl!

Reply:
As it is, ursh returns only the shortcut. Makes sense to also return its `BASE_URL`.